app/models/membership_application.rb

```diff
@@ -110,7 +110,7 @@ def marked_ready_for_review=(value)
 
 
   def swedish_organisationsnummer
-    errors.add(:company_number, "#{self.company_number} är inte ett svenskt organisationsnummer") unless Orgnummer.new(self.company_number).valid?
+    errors.add(:company_number, :invalid, company_number: self.company_number) unless errors.include?(:company_number) || Orgnummer.new(self.company_number).valid?
```

Comment:
Nice change - why complain about invalid company number if it missing or does not even have enough characters?